### PR TITLE
CP-220 : update manifest v3

### PR DESCRIPTION
**Pull request by LoganPB**
## 📝 What is this PR

This PR updates the version of the manifest to upgrade from V2 to V3. The new version is required by Google Chrome starting next year.

-   [Specs](https://phantombuster.atlassian.net/browse/CP-220)

## 🏭 How does it work

Google has a guide to migrate from V2 to V3 [https://developer.chrome.com/docs/extensions/mv3/mv3-migration/](https://developer.chrome.com/docs/extensions/mv3/mv3-migration/). The manifest and some files of the code has been updated. There's also the change from "webextension-polyfill-ts" to "@types/webextension-polyfill" because the first one is deprecated now.

## 🗒 Notes
This version is not functional on Firefox at the moment. Firefox should support Manifest V3 at the end of the year "normally".

## 🧪 QA Process

To test the extension :

npm i
npm run clean
npm run build:dev
On Google Chrome, open chrome://extensions/
Enable the Developer mode
Click on Load Unpacked
Select the dev-build folder
The extension should be activated in the browser (don't forget to disable the official extension)

### ✅ Positive testing (should succeed)

-  To test the update :

**Test when the user is already connected to the service:**
- Use a phantom
- Click on the "Connect to XXXX" button to get the cookie
- The phantom should work

**Test when the user is not connected to the service:**
- Use a phantom
- Click on the "Connect to XXXX" button. You should have a new tab opened
- Log in
- You are redirected to the PhantomBuster page

**Test to click the "Connect to XXXX" with Shift pushed**
- Use a phantom
- Click on the "Connect to XXXX" button with Shift pushed
- Nothing should change

## 👌 To check

-   [ ] There's no regression
-   [ ] Coding guidelines are followed ([Contributing](https://github.com/phantombuster/engineering-docs/blob/master/contributing.md), [Testing](https://github.com/phantombuster/engineering-docs/blob/master/testing.md))

### CODE REVIEW

**Comment by Hatscat** [comment]
> ## ⚠️ The Manifest was updated! Be very careful about this! ([more infos here](https://github.com/phantombuster/web-browser-extension#%EF%B8%8F-manifest-update-warning))
> ```diff
> 2c2
> < 	"manifest_version": 3,
> ---
> > 	"manifest_version": 2,
> 13c13
> < 		"service_worker": "background.js"
> ---
> > 		"scripts": ["background.js"]
> 15c15
> < 	"action": {
> ---
> > 	"browser_action": {
> 28,29c28,31
> < 	"permissions": ["notifications", "tabs", "cookies"],
> < 	"host_permissions": [
> ---
> > 	"permissions": [
> > 		"notifications",
> > 		"tabs",
> > 		"cookies",
> 46,51c48
> < 	"web_accessible_resources": [
> < 		{
> < 			"resources": ["assets/*"],
> < 			"matches": ["<all_urls>"]
> < 		}
> < 	]
> ---
> > 	"web_accessible_resources": ["assets/*"]
> ```

I have the impression that the diff is reversed ^^"

File: package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
 	"license": "MIT",
 	"devDependencies": {
 		"@types/node": "^14.14.25",
+		"@types/webextension-polyfill": "^0.9.1",
```
Comment (Hatscat) [inline package.json:26]: I think you forgot to commit the lock file, there are some diffs inside it when running `npm i`:
image

File: manifest.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-	"manifest_version": 2,
+	"manifest_version": 3,
 	"name": "PhantomBuster",
 	"version": "1.1.6.8",
```
Comment (Hatscat) [inline manifest.json:4]: plz bump the version, at least a patch I think (so you have to update `beta-updates.json`, `manifest.beta.json` and `package.json` files too
```suggestion
	"version": "1.1.7.0",
```

**Review by Hatscat — CHANGES_REQUESTED** [review]
it works well for me 🙂

File: package.json (hunk 1 above)
Comment (LoganPB) [reply package.json:26]: Fixed 👍

**Review by Hatscat — APPROVED** [review]
LGTM, GG 👍

**Review by RiahiKarim — COMMENTED** [review]
Great job @LoganPB. Thank you for the detailed QA process.

Code LGTM 💯 .

The idea of having a code base on the main branch that doesn't build on firefox makes me uneasy, maybe we should create a `epic/manifest-v3` branch for now and wait for firefox to fully support manifest v3.

@Hatscat WDYT?

**Comment by Hatscat** [comment]
> Great job @LoganPB. Thank you for the detailed QA process.
> 
> Code LGTM 💯 .
> 
> The idea of having a code base on the main branch that doesn't build on firefox makes me uneasy, maybe we should create a `epic/manifest-v3` branch for now and wait for firefox to fully support manifest v3.
> 
> @Hatscat WDYT?

yes good idea @RiahiKarim! we still could deploy from the epic, the only thing to think is to manually create sentry release because it's done at CI time when merging in beta

File: src/shared/inject.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
-import { browser } from "webextension-polyfill-ts"
+import * as browser from "webextension-polyfill"
 
 const headElement = document.head || document.documentElement
 
 export const injectJsFile = (fileName: string) => {
```
[the remark this answers is not in the document]
Comment (LoganPB) [reply src/shared/inject.ts]: Fixed 👍

**Review by RiahiKarim — APPROVED** [review]
LGTM 💯

**Pull request merged**